yarn-project/simulator/src/public/enqueued_call_side_effect_trace.test.ts

```diff
@@ -94,28 +60,17 @@ describe('Enqueued-call Side Effect Trace', () => {
     trace = new PublicEnqueuedCallSideEffectTrace(startCounter);
   });
 
-  const toVMCircuitPublicInputs = (trc: PublicEnqueuedCallSideEffectTrace) => {
-    return trc.toVMCircuitPublicInputs(
-      constants,
-      createPublicCallRequest(avmEnvironment),
-      startGasLeft,
-      endGasLeft,
-      transactionFee,
-      avmCallResults,
-    );
-  };
-
   it('Should trace storage reads', () => {
     const leafPreimage = new PublicDataTreeLeafPreimage(slot, value, Fr.ZERO, 0n);
     trace.tracePublicStorageRead(address, slot, value, leafPreimage, Fr.ZERO, []);
     expect(trace.getCounter()).toBe(startCounterPlus1);
 
-    const expectedArray = PublicValidationRequests.empty().publicDataReads;
     const leafSlot = computePublicDataTreeLeafSlot(address, slot);
-    expectedArray[0] = new PublicDataRead(leafSlot, value, startCounter /*contractAddress*/);
+    const read = new PublicDataRead(leafSlot, value, startCounter /*contractAddress*/);
 
-    const circuitPublicInputs = toVMCircuitPublicInputs(trace);
-    expect(circuitPublicInputs.validationRequests.publicDataReads).toEqual(expectedArray);
+    const sideEffects = trace.getSideEffects();
+    expect(sideEffects.publicDataReads.length).toEqual(1);
+    expect(sideEffects.publicDataReads[0]).toEqual(read);
```

Comment:
It's better to just do something like
`expect(sideEffects.publicDataReads).toEqual([read]);`
instead of expecting length and then indexing

same in other tests